p2p/background/kad_discovery_baseline_test.go

```diff
@@ -0,0 +1,128 @@
+package background
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	libp2pCrypto "github.com/libp2p/go-libp2p/core/crypto"
+	libp2pHost "github.com/libp2p/go-libp2p/core/host"
+	libp2pPeer "github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/pocket/runtime/defaults"
+)
+
+func TestLibp2pKademliaPeerDiscovery(t *testing.T) {
+	ctx := context.Background()
+
+	addr1, host1, _ := setupHostAndDiscovery(t, ctx, defaults.DefaultP2PPort, nil)
+
+	bootstrapAddr, err := multiaddr.NewMultiaddr(fmt.Sprintf("%s/p2p/%s", addr1, host1.ID().String()))
+	require.NoError(t, err)
+
+	addr2, host2, _ := setupHostAndDiscovery(t, ctx, defaults.DefaultP2PPort+1, bootstrapAddr)
+	addr3, host3, _ := setupHostAndDiscovery(t, ctx, defaults.DefaultP2PPort+2, bootstrapAddr)
+
+	expectedPeerIDs := []libp2pPeer.ID{host1.ID(), host2.ID(), host3.ID()}
+
+	// TECHDEBT: consider using `host.ConnManager().Notifee()` to avoid sleeping here
+	// delay assertions for 500ms
+	time.Sleep(time.Millisecond * 500)
+
+	// assert that host2 has host3 in its peerstore
+	host2DiscoveredAddrs := host2.Peerstore().Addrs(host3.ID())
+	require.Lenf(t, host2DiscoveredAddrs, 1, "did not discover host3")
+	require.Equalf(t, addr3.String(), host2DiscoveredAddrs[0].String(), "did not discover host3")
+	require.ElementsMatchf(t, expectedPeerIDs, host2.Peerstore().Peers(), "host2 peer IDs don't match")
+
+	// assert that host3 has host2 in its peerstore
+	host3DiscoveredHost2Addrs := host3.Peerstore().Addrs(host2.ID())
+	require.Lenf(t, host3DiscoveredHost2Addrs, 1, "host3 did not discover host2")
+	require.Equalf(t, addr2.String(), host3DiscoveredHost2Addrs[0].String(), "host3 did not discover host2")
+	require.ElementsMatchf(t, expectedPeerIDs, host3.Peerstore().Peers(), "host3 peer IDs don't match")
+
+	// add another peer to network...
+	addr4, host4, _ := setupHostAndDiscovery(t, ctx, defaults.DefaultP2PPort+3, bootstrapAddr)
+	expectedPeerIDs = append(expectedPeerIDs, host4.ID())
+
+	// TECHDEBT: consider using `host.ConnManager().Notifee()` to avoid sleeping here
```

Comment:
Ditto as above as to why we need a sleep.

Maybe move it to a constant ith a comment.

```go
// ...
const dhtUpdateSleepTime := time.Millisecond * 500
```